src/item.cpp

```diff
@@ -209,6 +209,29 @@ item::item( const itype *type, time_point turn, solitary_tag )
 item::item( const itype_id &id, time_point turn, solitary_tag tag )
     : item( find_type( id ), turn, tag ) {}
 
+item::item( const recipe *rec, long qty, t_item_vector items )
+    : item( "craft", calendar::turn, qty )
+{
+    making = rec;
+    components = items;
+
+    // Process this item to apply rot to components
+    active = true;
+
+    for( const item &it : components ) {
+        if( it.has_flag( "HIDDEN_POISON" ) ) {
+            set_flag( "HIDDEN_POISON" );
```

Comment:
This approach isn't particularly robust: imagine adding new flags or removing them. Essentially this snipped is code duplication with all its drawbacks. I'd tweak `item::has_flag` instead.

Reply:
The reason I did it this way is that I wasn't sure if it made sense for all flags to be transferred and didn't want to cause any potential strange behavior.  Will take another look at this.